File: problem10.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

#define maxn 2000009
bool primes[maxn];

void sieve(){
	primes[0]=0;
	primes[1]=0;
	for(int i=2;i<maxn;i++){
		if(primes[i]){
			for(int j=2*i;j<maxn;j+=i)
				primes[j]=0;
		}
	}
}

int main(){
	fill(primes,primes+maxn,1);
	sieve();
	ll ans=0;
	for(int i=1;i<=2000000;i++){
		if(primes[i])ans+=i;
	}
	cout<<ans<<endl;

}

File: solutions/problem10.py
primes = [True for i in range(2000001)]

def sieve():
	primes[0], primes[1] = False, False
	for i in range(2, 2000001):
		if(primes[i]):
			for j in range(2*i,2000001,i):
				primes[j]=False

ans=0
sieve()
for i in range(1,2000001):
	if(primes[i]):		
		ans = ans+i
print(ans)
